### PR TITLE
Support references on dataElements

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Related Issue(s)
- #223 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

### CODE REVIEW

**Comment by acn-sbuad** [comment]
Logic related to the code smell is being considered moved into a new service in another PR, so leaving it as is for now.

File: src/Storage/Helpers/DataElementHelper.cs [hunk 1]
```diff
@@ -38,13 +40,30 @@ public static DataElement CreateDataElement(string dataType, List<Guid> refs, In
                 LastChanged = creationTime,
                 Size = fileSize,
                 Refs = refs,
-            };            
+            };
+
+            if (generatedFromId != null && generatedFromId.Any())
```
Comment (SandGrainOne) [inline src/Storage/Helpers/DataElementHelper.cs:45]: Use the List.Count before IEnumeration.Any().

Comment (SandGrainOne) [reply src/Storage/Helpers/DataElementHelper.cs:45]: I keep forgetting that Any is smart and can detect that the underlying type is a List. Your choice.

Comment (SandGrainOne) [reply src/Storage/Helpers/DataElementHelper.cs:45]: Last comment. It's technically unnecessary to check for elements before a foreach. It's built in.

**Review by SandGrainOne — APPROVED** [review]

File: src/Storage/Helpers/DataElementHelper.cs (hunk 1 above)
Comment (acn-sbuad) [reply src/Storage/Helpers/DataElementHelper.cs:45]: but we avoid instantiating the references list if we don't have any references to include, so I guess the check still makes sense?

File: src/Storage/Controllers/DataController.cs [hunk 2]
```diff
@@ -445,6 +454,7 @@ public async Task<ActionResult<DataElement>> OverwriteData(int instanceOwnerPart
             {
                 { "/locked", dataElement.Locked },
                 { "/refs", dataElement.Refs },
+                { "/references", dataElement.References },
```
[the remark this answers is not in the document]
Comment (acn-sbuad) [reply src/Storage/Controllers/DataController.cs:457]: Limit is 10, så we still have 3-4 to go. But good point, I was not aware of the limitation.

File: src/Storage/Helpers/DataElementHelper.cs (hunk 1 above)
Comment (SandGrainOne) [reply src/Storage/Helpers/DataElementHelper.cs:45]: Yeah, keep it.

With that being said there is another side to that discusion we should take one day. The nullable discussion. The rules say that a List reference should never be null. Just nitpicking. 😄

**Pull request merged**